pkg/api/types.go

```diff
@@ -123,10 +123,11 @@ type PublicKey struct {
 
 // WindowsProfile represents the windows parameters passed to the cluster
 type WindowsProfile struct {
-	AdminUsername string            `json:"adminUsername"`
-	AdminPassword string            `json:"adminPassword"`
-	ImageVersion  string            `json:"imageVersion"`
-	Secrets       []KeyVaultSecrets `json:"secrets,omitempty"`
+	AdminUsername         string            `json:"adminUsername"`
+	AdminPassword         string            `json:"adminPassword"`
+	ImageVersion          string            `json:"imageVersion"`
+	WindowsImageSourceURL string            `json:"windowsImageSourceURL"`
```

Comment:
I would recommend apply to whole cluster (all agent pools). or per agent pool, but not both.  Which matches your scenario better?

Reply:
Whole cluster.  I beleive that is doing that. Am I missing something?

Reply:
If "whole cluster" is your scenario, I think you can completely remove the per node pool change you made in parts/agentparams.t.